crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -248,3 +249,28 @@ impl<'a> ReFunc<'a> {
         })
     }
 }
+
+/// Recursively try to resolve `name` to an [`ExprStringLiteral`] in `semantic`.
+fn resolve_name<'a, 'b>(
```

Comment:
Nit: Maybe rename to `resolve_pattern` or `resolve_string_literal` because it isn't returning a name.